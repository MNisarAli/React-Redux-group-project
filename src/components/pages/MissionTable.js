import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchedMissions, joinMission } from '../../redux/missions/mission';
import Badge from '../Badge';
import '../../style.css';

const MissionsTable = () => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.missions);
  const { status, missions } = data;

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchedMissions());
    }
  }, [dispatch, status]);

  const handleClick = (mission) => {
    dispatch(joinMission(mission));
  };

  return (
    <table>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th aria-label="empty-th" />
        </tr>
      </thead>
      <tbody>
        {
          missions.map((mission) => {
            const {
              id, name, description, isReserved,
            } = mission;
            return (
              <tr key={id}>
                <td>{name}</td>
                <td>{description}</td>
                <td>
                  <Badge
                    text={isReserved ? 'Active member' : 'Not a member'}
                    active={isReserved}
                  />
                </td>
                <td>
                  <button
                    type="button"
                    className={`${isReserved && 'btn-active'}`}
                    onClick={() => handleClick(mission)}
                  >
                    {isReserved ? 'Leave Mission' : 'Join Mission'}
                  </button>
                </td>
              </tr>
            );
          })
        }
      </tbody>
    </table>
  );
};

export default MissionsTable;
