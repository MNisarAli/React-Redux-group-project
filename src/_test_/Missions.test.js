import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MissionTable from '../components/pages/MissionTable';
import store from '../redux/configureStore';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}><MissionTable /></Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
