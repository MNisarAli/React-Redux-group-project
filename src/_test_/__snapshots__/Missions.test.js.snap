// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th
        aria-label="empty-th"
      />
    </tr>
  </thead>
  <tbody />
</table>
`;
